src/models/extensions/defaultExtension.ts

```diff
@@ -0,0 +1,8 @@
+import { FileFormat } from './fileFormat';
+
+export interface IDefaultExtension {
+  icon: string; // name of the icon.
+  format: FileFormat | string; // format of the icon
+  _custom?: boolean; // user customization: flag for the icon generator.
```

Comment:
We could use another `interface` with just this property but since the users can enter whatever property they want in their settings (remember they don't rely on Typescript) they could easily inject it and the code complexity it would introduce I don't think is worth the effort. The good news is that nothing would happen anyway as the value of this property will get overwritten by the `merge functionality`.